Source/SG/SG.cs

```diff
@@ -365,7 +366,8 @@ public static Waveform GetWaveformParametersByName(NIRfsg rfsgHandle, string wav
             NIRfsgPlayback.RetrieveWaveformSampleRate(rfsgPtr, waveformName, out waveform.SampleRate);
             NIRfsgPlayback.RetrieveWaveformBurstStartLocations(rfsgPtr, waveformName, ref waveform.BurstStartLocations);
             NIRfsgPlayback.RetrieveWaveformBurstStopLocations(rfsgPtr, waveformName, ref waveform.BurstStopLocations);
-            waveform.IdleDurationPresent = waveform.BurstStartLocations?.Length > 0;
+            NIRfsgPlayback.RetrieveWaveformSize(rfsgPtr, waveformName, out int waveformSize);

```

Comment:
I found this bug while writing the LV code. With the new waveform object getting returned we have to get the waveform size manually else the IdleDurationPresent flag would have always been true since `waveformSize - 1` would have always returned -1.

Reply:
Oh, nice catch. This is definitely a drawback of this function; it is called `GetWaveformParameters` rather than `GetWaveform` since the waveform itself isn't actually retrieved, but it still returns a `Waveform` struct with the `WaveformData` field. That's prone to error if someone doesn't understand this fact. However, this function plays a very important role for automation so I don't quite know how to get around it. I have just hoped that no one would have any reason to try retrieving the actual waveform data by name...